solr/core/src/java/org/apache/solr/response/PrometheusResponseWriter.java

```diff
@@ -118,13 +108,41 @@ public static void toPrometheus(
         (metricName, metric) -> {
           try {
             Metric dropwizardMetric = dropwizardMetrics.get(metricName);
-            solrPrometheusCoreExporter.exportDropwizardMetric(dropwizardMetric, metricName);
+            exporter.exportDropwizardMetric(dropwizardMetric, metricName);
           } catch (Exception e) {
             // Do not fail entirely for metrics exporting. Log and try to export next metric
             log.warn("Error occurred exporting Dropwizard Metric to Prometheus", e);
           }
         });
 
-    consumer.accept(solrPrometheusCoreExporter);
+    consumer.accept(exporter);
+  }
+
+  public static SolrPrometheusExporter getExporterType(String registryName) {
+    String coreName;
+    boolean cloudMode = false;
+    String[] rawParsedRegistry = registryName.split("\\.");
+    List<String> parsedRegistry = new ArrayList<>(Arrays.asList(rawParsedRegistry));
+
+    switch (parsedRegistry.get(1)) {
+      case ("core"):
```

Comment:
These parenthesis in your case statements are pointless; not sure I've ever seen that